[SEXP-4] Fix ECS ARN parsing for GovCloud and China regions

hmmferreira opened the pull request:
### What does this PR do?

Fixes ECS ARN parsing to support AWS GovCloud (`aws-us-gov`) and China (`aws-cn`) partitions in addition to the standard `aws` partition. Also consolidates duplicate implementations to ensure the fix applies consistently across the entire codebase.

### Motivation

The `ParseRegionAndAWSAccountID` function in `pkg/util/ecs/ecs.go` only accepted the `"aws"` partition, causing ECS metadata extraction to fail for customers running the Datadog Agent in AWS GovCloud or China regions. This resulted in empty region and account ID values, breaking ECS monitoring for these customers.

AWS has three partitions:
- `aws` - Commercial regions
- `aws-us-gov` - GovCloud regions
- `aws-cn` - China regions

A duplicate implementation existed in `comp/core/workloadmeta/collectors/util/ecs_util.go` with the same bug. This PR consolidates all ECS ARN parsing to use the fixed implementation in `pkg/util/ecs`, ensuring consistent behavior across the agent.

### Describe how you validated your changes

- Added comprehensive table-driven tests covering all three AWS partitions
- Added tests for edge cases (invalid partitions, malformed ARNs)
- All tests pass: `go test -tags docker ./pkg/util/ecs -v` and `go test -tags docker ./comp/core/workloadmeta/collectors/util -v`
- Verified all 5 new test cases pass:
  - Standard aws partition
  - aws-us-gov partition (GovCloud)
  - aws-cn partition (China)
  - Invalid partition returns empty (as expected)
  - Malformed ARN returns empty (as expected)
- Verified consolidation: removed duplicate implementation and updated all call sites (v1parser, v2parser, ecs_util)

### Additional Notes

This is a backwards-compatible change that maintains existing behavior for standard AWS regions while extending support to GovCloud and China regions. The consolidation removes 22 lines of duplicate code and ensures the fix applies to all ECS metadata collection paths.

#### Tickets
- Support ticket: [SEXP-4](https://datadoghq.atlassian.net/browse/SEXP-4) 
- Internal ticket:  [EXP-139](https://datadoghq.atlassian.net/browse/EXP-139)

[SEXP-4]: https://datadoghq.atlassian.net/browse/SEXP-4?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

hmmferreira committed:
Bump agent payload to v5.0.160

hmmferreira committed:
Merge remote-tracking branch 'origin/main'

# Conflicts:
#	comp/otelcol/ddflareextension/impl/go.mod
#	comp/otelcol/ddflareextension/impl/go.sum
#	comp/otelcol/logsagentpipeline/go.mod
#	comp/otelcol/logsagentpipeline/go.sum
#	comp/otelcol/logsagentpipeline/logsagentpipelineimpl/go.mod
#	comp/otelcol/logsagentpipeline/logsagentpipelineimpl/go.sum
#	comp/otelcol/otlp/components/connector/datadogconnector/go.mod
#	comp/otelcol/otlp/components/connector/datadogconnector/go.sum
#	comp/otelcol/otlp/components/exporter/datadogexporter/go.mod
#	comp/otelcol/otlp/components/exporter/datadogexporter/go.sum
#	comp/otelcol/otlp/components/exporter/serializerexporter/go.mod
#	comp/otelcol/otlp/components/exporter/serializerexporter/go.sum
#	go.mod
#	go.sum
#	pkg/logs/pipeline/go.mod
#	pkg/logs/pipeline/go.sum
#	pkg/logs/processor/go.mod
#	pkg/logs/processor/go.sum
#	pkg/process/util/api/go.mod
#	pkg/process/util/api/go.sum
#	pkg/serializer/go.mod
#	pkg/serializer/go.sum
#	test/fakeintake/go.mod
#	test/fakeintake/go.sum
#	test/new-e2e/go.mod
#	test/new-e2e/go.sum
#	test/otel/go.mod
#	test/otel/go.sum

hmmferreira committed:
Merge remote-tracking branch 'origin/main'

hmmferreira committed:
[SEXP-4] Fix ECS ARN parsing for GovCloud and China regions

Accept aws-us-gov and aws-cn partitions in addition to aws partition.
Previously only "aws" was accepted, breaking ECS metadata extraction
for GovCloud and China customers.

hmmferreira committed:
Merge branch 'main' into hugo.ferreira/sexp-4

hmmferreira committed:
Add release note for ECS GovCloud/China fix

hmmferreira committed:
Rename release note to follow naming convention

stzou reviewed (commented):
Looks like we need to update `comp/core/workloadmeta/collectors/util/ecs_util.go` as well since `ParseRegionAndAWSAccountID` is duplicated there. Perhaps we can consolidate them so this logic isn't duplicated.

hmmferreira committed:
Consolidate duplicate ECS ARN parsing to fix GovCloud/China everywhere

Remove duplicate ParseRegionAndAWSAccountID from workloadmeta/collectors/util.
All ECS ARN parsing now uses the fixed version in pkg/util/ecs, ensuring
GovCloud and China partition support applies consistently across the codebase.

hmmferreira committed:
Merge branch 'main' into hugo.ferreira/sexp-4

hmmferreira committed:
Merge branch 'main' into hugo.ferreira/sexp-4

L3n41c commented on comp/core/workloadmeta/collectors/util/ecs_util_test.go:
```diff
@@ -10,19 +10,20 @@ package util
 	"github.com/stretchr/testify/require"
 )
 
 func TestParseRegionAndAWSAccountID(t *testing.T) {
```
Since this test tests the `ParseRegionAndAWSAccountID` that has been consolidated with the one of `pkg/util/ecs`, wouldn’t it make more sense to also move those tests to `pkg/util/ecs`?

In fact, it looks like those cases are already covered by the `TestGetRegionAndAWSAccountID` of `‎pkg/util/ecs` so that this test looks not needed anymore.

hmmferreira replied:
Nicely spotted! Thx!
Already addressed 👍

L3n41c reviewed (commented):
Since there’s no `ParseRegionAndAWSAccountID` function in the `‎comp/core/workloadmeta/collectors/util` package anymore (there’s only one in `pkg/util/ecs`, I think we can remove the `TestParseRegionAndAWSAccountID` test from `comp/core/workloadmeta/collectors/util` and keep only the one in `pkg/util/ecs`.

It looks weird that the unit tests of a package tests functions from another package.

hmmferreira committed:
Remove redundant ParseRegionAndAWSAccountID test after consolidation

Move unique test case (invalid account ID length) to pkg/util/ecs/ecs_test.go
and remove the now-redundant TestParseRegionAndAWSAccountID from
comp/core/workloadmeta/collectors/util/ecs_util_test.go.

After consolidating ParseRegionAndAWSAccountID to pkg/util/ecs, the test
in ecs_util_test.go became redundant. The comprehensive tests in
pkg/util/ecs/ecs_test.go now cover all cases including the unique edge
case for invalid account ID length (3 digits instead of 12).
